### PR TITLE
[OV2.0] Preprocessing documentation

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
 - Preprocessing overview
 - Preprocessing details
 - Layout overview
 - Updated some comments in header files

### Tickets:
 - 62182

### CODE REVIEW

**Comment by yury-gorbachev** [comment]
This document is great start, terrific work.

File: docs/OV_Runtime_UG/layout_overview.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    - Apply [preprocessing](./preprocessing_overview.md) steps, like subtract means, divide by scales, resize image, convert RGB<->BGR
+    - Set/get batch for a model
+ - Same operations, used during model conversion phase, see [Model Optimizer model conversion](../MO_DG/prepare_model/convert_model/Converting_Model.md)
+ - Improve readability of a model's input and output
+
+## Layout syntax
+
+### Short
+The easiest way is to fully specify each dimension with one alphabetical letter
+- C++
```
Comment (ilyachur) [inline docs/OV_Runtime_UG/layout_overview.md:22]: Could you use tabs for C++ and Python documentation?
Example: https://github.com/openvinotoolkit/openvino/pull/10187/files#diff-bc0e41845309d0fe7133a7882677bb0cdd304429101f5b10eca781d7e7bc40bcR15-R29

File: docs/OV_Runtime_UG/layout_overview.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    @snippet snippets/ov_layout.cpp ov:layout:simple
+- Python
+  ```
+  from openvino.runtime import Layout
```
Comment (ilyachur) [inline docs/OV_Runtime_UG/layout_overview.md:26]: Could you use snippets for python code?

Comment (nosovmik) [reply docs/OV_Runtime_UG/layout_overview.md]: Done

File: docs/OV_Runtime_UG/layout_overview.md (hunk 1 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/layout_overview.md]: Done, thanks for pointing to this PR

File: docs/OV_Runtime_UG/preprocessing_details.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+.. tab:: Python
+
+      .. doxygensnippet:: docs/snippets/ov_preprocessing.py
+         :language: python
+         :fragment: [ov:preprocess:input_1]
+
+@endsphinxdirective
+
+In general, when model has multiple inputs/outputs, each one can be addressed by name
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_details.md:25]: maybe `by tensor name ` ?

File: docs/OV_Runtime_UG/preprocessing_details.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+  * <code>ov::preprocess::PreProcessSteps::convert_layout()</code>
+  * <code>ov::preprocess::InputTensorInfo::set_layout()</code>
+  * <code>ov::preprocess::InputModelInfo::set_layout()</code>
+  * <code>ov::Layout</code>
+
+#### Resize image
+
+Resizing of image is a typical preprocessing step for computer vision tasks. With preprocessing API this step can also be integrated into execution graph and performed on target device.
+
+To resize the input image, it is needed to define 'H' and 'W' dimensions of [layout](./layout_overview.md)
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_details.md:196]: replace ' with ` ?

File: docs/OV_Runtime_UG/preprocessing_details.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+C++ references:
+* <code>ov::preprocess::ColorFormat</code>
+* <code>ov::preprocess::PreProcessSteps::convert_color</code>
+
+
+### Custom operations
+
+Preprocessing API also allows adding custom preprocessing steps into execution graph. Custom step is a function which accepts current 'input' node and returns new node after adding preprocessing step
+
+>**Note:** Custom preprocessing function shall only insert node(s) after input, it will be done during model compilation. This function will NOT be called during execution phase. This may look not trivial and require some knowledge of [OpenVINO™ operations](../ops/opset.md)
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_details.md:288]: this "note" is not rendered as expected. I still see `>`
In other places in this documentation - as well

File: docs/OV_Runtime_UG/preprocessing_overview.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+ - Preprocessing steps will be integrated into execution graph and will be performed on selected device (CPU/GPU/VPU/etc.) rather than always being executed on CPU. This will improve selected device utilization which is always good.
+
+## Preprocessing API
+
+Intuitively, Preprocessing API consists of the following parts:
+ 1. 	**Tensor:** Declare user's data format, like shape, [layout](./layout_overview.md), precision, color format of actual user's data
+ 2. 	**Steps:** Describe sequence of preprocessing steps which need to be applied to user's data
+ 3. 	**Model:** Specify Model's data format. Usually, precision and shape are already known for model, only additional information, like [layout](./layout_overview.md) can be specified
+
+>**Note:** All model's graph modification shall be performed after model is read from disk and **before** it is being loaded on actual device. See also [OpenVINO™ Common Inference pipeline](../migration_ov_2_0/docs/common_inference_pipeline.md)
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_overview.md:41]: it should refer to pipeline in integration guide, not in transition to OV 2.0 I suppose

File: docs/OV_Runtime_UG/preprocessing_overview.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+      .. doxygensnippet:: docs/snippets/ov_preprocessing.py
+         :language: python
+         :fragment: [ov:preprocess:create]
+
+@endsphinxdirective
+
+### Declare user's data format
+
+To address particular input of model/preprocessor, use `PrePostProcessor::input(input_name)` method
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_overview.md:65]: should we add `ov::` to have auto-generated link in docs ?

File: docs/OV_Runtime_UG/preprocessing_overview.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+      .. doxygensnippet:: docs/snippets/ov_preprocessing.py
+         :language: python
+         :fragment: [ov:preprocess:steps]
+
+@endsphinxdirective
+
+Here:
+ - Convert U8 to FP32 precision
+ - Convert current color format (BGR) to RGB
+ - Resize to model's height/width. **Note** that if model accepts dynamic size, e.g. {?, 3, ?, ?}, `resize` will not know how to resize the picture, so in this case you should specify target height/width on this step. See also [TBD]
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/preprocessing_overview.md:136]: can we fix [TBD] ?

Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_overview.md]: Fixed

File: docs/OV_Runtime_UG/preprocessing_overview.md (hunk 7 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_overview.md]: Added

File: docs/OV_Runtime_UG/preprocessing_overview.md (hunk 6 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_overview.md]: I did such change, but appropriate page will exist only after merge of https://github.com/openvinotoolkit/openvino/pull/10187

File: docs/OV_Runtime_UG/preprocessing_details.md (hunk 5 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_details.md]: Should be fixed now

File: docs/OV_Runtime_UG/preprocessing_details.md (hunk 4 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_details.md]: Done

File: docs/OV_Runtime_UG/preprocessing_details.md (hunk 3 above)
Comment (nosovmik) [reply docs/OV_Runtime_UG/preprocessing_details.md]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**